lib/retrieval-stats.js

```diff
@@ -95,9 +95,9 @@ export const buildRetrievalStats = (measurements, telemetryPoint) => {
 const addHistogramToPoint = (point, fieldNamePrefix, values) => {
   const count = values.length
   if (count < 1) return
-  values.sort()
+  values.sort((a, b) => a - b)
   point.intField(`${fieldNamePrefix}_min`, values[0])
-  point.intField(`${fieldNamePrefix}_mean`, values.reduce((sum, v) => sum + v, 0) / count)
+  point.intField(`${fieldNamePrefix}_mean`, values.reduce((sum, v) => sum + BigInt(v), 0n) / BigInt(count))
```

Comment:
This change is not strictly necessary, but I think it can improve the precision of this metric.